src/libraries/Microsoft.PowerFx.Interpreter/Marshal/TypeCoercionProvider.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Globalization;
+using System.Threading;
+using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Functions;
+using Microsoft.PowerFx.Types;
+using static Microsoft.PowerFx.Functions.Library;
+
+namespace Microsoft.PowerFx.Interpreter.Marshal
+{
+    /// <summary>
+    /// Convert value to target format.
+    /// </summary>
+    internal static class TypeCoercionProvider
```

Comment:
>TypeCoercionProvider [](http://example.com/codeflow?start=26&length=20)

We can expose this publicly on FormulaValue via extensions method:

        public static bool TryCoerceTo(this FormulaValue fv, FormulaType targetType, out FormulaValue newValue)
        {
            // calls your impl 
        }

Reply:
updated this.

Reply:
The class needs to be public for this to be publicly callable. 

And it should be in the root namespace (ignore the file path)
 Microsoft.PowerFx

Reply:
Updated